api/src/main/java/com/cloud/template/TemplateApiService.java

```diff
@@ -56,9 +56,9 @@
 
     VirtualMachineTemplate prepareTemplate(long templateId, long zoneId, Long storageId);
 
-    boolean detachIso(long vmId);
+    boolean detachIso(long vmId, boolean forced);
```

Comment:
can we define a default value for these boolean variables? this way you can avoid doing `forced != null`

Reply:
it's passed from API layer - which handles this and ensures that this is either false or true. Furthermore, `boolean` is a primitive type and cannot be `null`.